Draft: Remove some usage of unittest

maringuu opened the pull request:
In the same spirit as #922 and #923 this pr aims to remove the usage of unittest in our codebase.
While the other PRs are quite complex this one is pretty straight forward and can probably be merged first.

With that beeing said this is still wip.

maringuu commented:
Ready for review.

maringuu commented:
https://github.com/fkie-cad/FACT_core/actions/runs/3702060797

maringuu commented:
Read for review once the CI passes.
https://github.com/fkie-cad/FACT_core/actions/runs/3702858426

maringuu commented on docsrc/conf.py:
```diff
@@ -72,6 +73,7 @@
     'pydantic',
     'pymongo',
     'pytest',
+    'quart',
```
I don't know why this is needed but building the docs complained about this.

maringuu commented:
I force pushed once again because I frogot to fixup one commit. So the code [didn't change](https://github.com/fkie-cad/FACT_core/compare/ea504a6b3b77d33368f64d48fafcf88a35a38c5d..49592785cf41a6719ceeb8019ec5ac3cbc4ca5f3)

jstucke commented on src/test/unit/web_interface/test_app_jinja_filter.py:
```diff
@@ -1,43 +1,42 @@
-        super().setup()
-        self.filter = FilterClass(self.frontend.app, '', self.frontend.db)
+@pytest.fixture
+def filter(web_frontend):
```
filter is actually a built-in python class so we should use a different name

jstucke reviewed (changes requested):
Looks good apart from some small stuff

maringuu commented on src/test/conftest.py:
```diff
@@ -1,9 +1,39 @@
 from test.common_helper import clear_test_tables, setup_test_tables
 
 
+def merge_markers(request, name: str, dtype: Type) -> TypeVar('T'):
```
Good catch. No idea if this would have worked.

But your solution is better anyways since it shows the relation between `dytpe` and the return value.

maringuu commented on src/test/integration/intercom/test_task_communication.py:
```diff
@@ -48,116 +52,107 @@ def get_binary_and_file_name(uid: str) -> tuple[bytes, str]:
+        monkeypatch.setattr(
+            'intercom.back_end_binding.BinaryService.get_binary_and_file_name', lambda *_: (b'test', 'test.txt')
+        )
+        monkeypatch.setattr('intercom.front_end_binding.generate_task_id', lambda *_: 'valid_uid_0.0')
```
If I'm not mistaken it is different in all four tests.
It looks similar but the functions that are patched are different ones, aren't they?

jstucke replied:
I was only talking about the `generate_task_id` patch and not the other ones (which are indeed different)

maringuu commented:
I addressed your review in several commits prefixed with `FIXUP:`.
After your acknolegment, I'll use these commits to fixup the respective commits that introduced the things that needed changes and force push one more time. Then I'll merge.

maringuu commented:
CI run: https://github.com/fkie-cad/FACT_core/actions/runs/3740098601
